### PR TITLE
Fix to ESApi connector_sync_job_update_stats() to ensure undefined metadata passed as {} when undefined

### PR DESCRIPTION
**Pull request by mattnowzari**
#### Pre-Review Checklist
- [x] this PR does NOT contain credentials of any kind, such as API keys or username/passwords (double check `config.yml.example`)
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Testes the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

## Related Pull Requests
https://github.com/elastic/connectors/pull/3110

### CODE REVIEW

File: tests/es/test_index.py [hunk 1]
```diff
@@ -408,13 +408,13 @@ async def test_es_api_connector_sync_job_claim():
 async def test_es_api_connector_sync_job_update_stats():
     sync_job_id = "sync_job_id_test"
     ingestion_stats = {"ingestion": "stat"}
-    metadata = {"meta": "data"}
```
Comment (jedrazb) [inline tests/es/test_index.py:411]: we can add additional test for the new edge case, wdyt?

Comment (mattnowzari) [reply tests/es/test_index.py:411]: Definitely! I'll add a new one

**Review by jedrazb — APPROVED** [review]
LGTM, ship if tested locally and sync completed and finished
